projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageDeployerV2.java

```diff
@@ -0,0 +1,376 @@
+
+  @Value("${datajobs.deployment.readOnlyRootFilesystem:}")
+  private boolean readOnlyRootFilesystem;
+
+  @Value("${datajobs.deployment.jobImagePullPolicy:IfNotPresent}")
+  private String jobImagePullPolicy;
+
+  private static final String VOLUME_NAME = "vdk";
+  private static final String VOLUME_MOUNT_PATH = "/vdk";
+  private static final String EPHEMERAL_VOLUME_NAME = "tmpfs";
+  private static final String EPHEMERAL_VOLUME_MOUNT_PATH = "/var/tmp";
+  private static final String KEYTAB_PRINCIPAL_ENV = "VDK_KEYTAB_PRINCIPAL";
+  private static final String KEYTAB_FOLDER_ENV = "VDK_KEYTAB_FOLDER";
+  private static final String KEYTAB_FILENAME_ENV = "VDK_KEYTAB_FILENAME";
+  private static final String ATTEMPT_ID = "VDK_ATTEMPT_ID";
+  private static final String BASE_CONFIG_FOLDER = "VDK_BASE_CONFIG_FOLDER";
+  private static final String VDK_TEMPORARY_WRITE_DIRECTORY = "VDK_TEMPORARY_WRITE_DIRECTORY";
+  private static final String TEMPORARY_WRITE_DIRECTORY_PATH = EPHEMERAL_VOLUME_MOUNT_PATH;
+  private static final String KEYTAB_FOLDER = "/credentials";
+
+  private final JobCredentialsService jobCredentialsService;
+  private final DataJobsKubernetesService dataJobsKubernetesService;
+  private final VdkOptionsReader vdkOptionsReader;
+  private final DataJobDefaultConfigurations defaultConfigurations;
+  private final DeploymentProgress deploymentProgress;
+  private final KubernetesResources kubernetesResources;
+  private final JobCommandProvider jobCommandProvider;
+  private final SupportedPythonVersions supportedPythonVersions;
+
+  /**
+   * Schedules for execution a data job at kubernetes cluster.
+   *
+   * @param dataJob the data job being deployed
+   * @param jobDeployment the deployment information necessary to exeucte the deployment
+   * @param sendNotification to sent or not notification (not it is sent only in case of failure,
+   *     caller is responsible for sending on success or in case of an exception)
+   * @param lastDeployedBy the name of the user that last deployed the data job
+   * @return true if it is successful and false if not.
+   */
+  public boolean scheduleJob(
+      DataJob dataJob,
+      JobDeployment jobDeployment,
+      Boolean sendNotification,
+      String lastDeployedBy,
+      Set<String> cronJobs) {
+    Validate.notNull(jobDeployment, "jobDeployment should not be null");
+    Validate.notNull(jobDeployment.getImageName(), "Image name is expected in jobDeployment");
+    log.info("Update cron job for data job {}", dataJob.getName());
+
+    try {
+      updateCronJob(dataJob, jobDeployment, lastDeployedBy, cronJobs);
+      return true;
+    } catch (ApiException e) {
+      return catchApiException(dataJob, jobDeployment, sendNotification, e);
+    }
+  }
+
+  private boolean catchApiException(
+      DataJob dataJob,
+      JobDeployment jobDeployment,
+      Boolean sendNotification,
+      ApiException apiException) {
+    if (apiException.getCode() == 422) {
```

Comment:
What does the 422 magic number denote?

Reply:
Unprocessable Entity - This is copied from JobImageDeployer.java.